### PR TITLE
Precise array Phpdoc for Request

### PR DESCRIPTION
**Pull request by VincentLanglet**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 for phpdoc
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

**Comment by fabpot** [comment]
That's for 6.2.

**Comment by VincentLanglet** [comment]
> That's for 6.2.

Kinda sad we need to wait 6 month for a phpdoc update, but I've updated my PR then to target 6.2 branch

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -1630,6 +1642,8 @@ public function getEncodings(): array
 
     /**
      * Gets a list of content types acceptable by the client browser in preferable order.
+     *
+     * @return string[] List of content types in preferable order
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/Request.php]: I would remove all return descriptions as they just say the same as the main description.

**Comment by VincentLanglet** [comment]
I have a psalm issue with the method:
```
    /**
     * Associates a format with mime types.
     *
     * @param string|string[] $mimeTypes The associated mime types (the preferred one must be the first as it will be used as the content type)
     */
    public function setFormat(?string $format, string|array $mimeTypes)
    {
        if (null === static::$formats) {
            static::initializeFormats();
        }

        static::$formats[$format] = \is_array($mimeTypes) ? $mimeTypes : [$mimeTypes];
    }
```

Seems like psalm is considering that when `$format` is null, the offset is 0. I tried and the offset is `''` which is a string.
I opened a psalm issue about it https://github.com/vimeo/psalm/issues/8063

Dunno if/how you want that I fix this for now @fabpot , I can use `$format ?? ''` as key for instance.
(But maybe  setting a null format should be deprecated)

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 2]
```diff
@@ -199,7 +199,7 @@ class Request
     protected $defaultLocale = 'en';
 
     /**
-     * @var array
+     * @var array<string, array<string>>
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpFoundation/Request.php]: `string[]` here also, or `list<string>` (but I prefer the former)

Comment (VincentLanglet) [reply src/Symfony/Component/HttpFoundation/Request.php]: There are the code 
```
static::$formats[$format] = \is_array($mimeTypes) ? $mimeTypes : [$mimeTypes];
```
so it's an array of array.

Did you mean you preferred `array<string, string[]>` over `array<string, array<string>>` ?

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpFoundation/Request.php]: > Did you mean you preferred array<string, string[]> over

yes :)

Comment (VincentLanglet) [reply src/Symfony/Component/HttpFoundation/Request.php]: Done :)

**Comment by VincentLanglet** [comment]
For the record the psalm failure is related to the psalm issue https://github.com/vimeo/psalm/issues/8063 which will be solved in the next psalm release.

But isn't weird to accept `null` as format here @nicolas-grekas (since it's used as an array key):
https://github.com/symfony/symfony/blob/89932de9f6d36ddb6d005983f3551b89f08fd0c8/src/Symfony/Component/HttpFoundation/Request.php#L1296-L1302

Then the getter only accept `string` 
https://github.com/symfony/symfony/blob/89932de9f6d36ddb6d005983f3551b89f08fd0c8/src/Symfony/Component/HttpFoundation/Request.php#L1256
or return `string`
https://github.com/symfony/symfony/blob/89932de9f6d36ddb6d005983f3551b89f08fd0c8/src/Symfony/Component/HttpFoundation/Request.php#L1279-L1286

So when you `set` with a null key, you have to `get` with `''`.

**Comment by nicolas-grekas** [comment]
I guess we accept null for historical reasons.

**Review by nicolas-grekas — APPROVED** [review]

**Review by maxhelias — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @VincentLanglet.

**Pull request closed without merging**